### PR TITLE
Chore/sync code to v7

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description

sync v6 to v7

### CODE REVIEW

**Comment by Theeraphat-Sorasetsakul** [comment]
@bualoy-napat  @wattachai-lseg 
Could you confirm specifically on `test-helpers/src/test-helpers.ts` file. It has many conflicts. And this is what I did.
 - Put _replaceWhitespace_ at the top of the file. Same position from v7.
 - Removed _keyboardEvent_ as I saw it remove since 6 months ago.
 - Put ResizeObserver code (the code begins with _before(function(..._ ) at the bottom of the file.

File: packages/core/src/elements/ResponsiveElement.ts [hunk 1]
```diff
@@ -31,14 +31,23 @@ export type ResizeEvent = CustomEvent<{
  * @returns {void}
  */
 const triggerResize = (entry: ResizeObserverEntry): void => {
-
-  const { inlineSize: width, blockSize: height } = entry.borderBoxSize[0];
+  let width: number;
+  let height: number;
+  /* istanbul ignore else */
```
Comment (wattachai-lseg) [inline packages/core/src/elements/ResponsiveElement.ts]: let's update to the syntax of v7 code coverage tool.

File: packages/test-helpers/src/test-helpers.ts [hunk 2]
```diff
@@ -78,3 +78,25 @@ export const isSafari = (version = undefined): boolean => { // Indicates if this
...
+// This module converts the error into a warning instead
+// Note that, mocha must be available in an imported context
+before(function () {
+  const originalOnError = window.onerror;
+  window.onerror = function (event, ...args) {
+    // Firefox: `ResizeObserver loop completed with undelivered notifications.`
+    // Chrome: `ResizeObserver loop limit exceeded`
+    // Safari: `ResizeObserver loop completed with undelivered notifications`
+    // Each browser logs a slightly different messages yet they all start with `ResizeObserver loop`
+    /* istanbul ignore else */
```
Comment (wattachai-lseg) [inline packages/test-helpers/src/test-helpers.ts]: let's update to the syntax of v7 code coverage tool.

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]
2 things

File: packages/core/src/elements/ResponsiveElement.ts (hunk 1 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/core/src/elements/ResponsiveElement.ts]: fixed.

File: packages/test-helpers/src/test-helpers.ts (hunk 2 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/test-helpers/src/test-helpers.ts]: fixed.

File: packages/core/src/elements/ResponsiveElement.ts [hunk 3]
```diff
@@ -31,14 +31,23 @@ export type ResizeEvent = CustomEvent<{
...
-
-  const { inlineSize: width, blockSize: height } = entry.borderBoxSize[0];
+  let width: number;
+  let height: number;
+  if (entry.borderBoxSize?.length > 0) {
+    width = entry.borderBoxSize[0].inlineSize;
+    height = entry.borderBoxSize[0].blockSize;
+  }
+  /* c8 ignore next 4 */
+  else {
```
Comment (wsuwt) [inline packages/core/src/elements/ResponsiveElement.ts:41]: Using line count isn't nice. Do we have any other way to accomplish this ?

Comment (Theeraphat-Sorasetsakul) [reply packages/core/src/elements/ResponsiveElement.ts:41]: Other option is start and stop. How do you think?

/* c8 ignore start */
...
/* c8 ignore stop */

Comment (Theeraphat-Sorasetsakul) [reply packages/core/src/elements/ResponsiveElement.ts:41]: Updated to use start and stop.

Comment (wsuwt) [reply packages/core/src/elements/ResponsiveElement.ts:41]: It's a lot better.

**Review by wattachai-lseg — APPROVED** [review]

**Review by wsuwt — APPROVED** [review]

**Pull request merged**